### PR TITLE
Improve Windows default folder

### LINKED ISSUE #50: Default desktop folder doesn't exist when using OneDrive sync.
I am synchronizing my user profile folders with OneDrive.
This results in the default %USERPROFILE%\Desktop folder not existing anymore.
OneDrive also changes the English name of the desktop folder to the localized name, in my case in dutch to Bureaublad.
%OneDrive% doesn't work in the shell. 

What I needed was:
```js
windows: {
    outputPath: execSync('[Environment]::GetFolderPath("Desktop")', {'shell':'powershell.exe'}).toString().trim(),
}
```

Maybe you can change the default to something like this since this is the way that works universally.
Source: https://stackoverflow.com/a/31747246/9242840 & https://stackoverflow.com/a/61219838/9242840

### PR DESCRIPTION
**Pull request by TheJaredWilcurt**
Fixes #50.

* [x] Feature complete
* [x] Version Bump
* [x] Update deps
* [x] 100% code coverage

### CODE REVIEW

**Comment by StoneyEagle** [comment]
You need to change the exec to execSync for it to work.
The regular exec works with a callback and doesn't return anything from the stdout.
You can also do something like:
```js
let desktopPath;
await new Promise((resolve, reject) => {
    exec('[Environment]::GetFolderPath("Desktop")', {'shell':'powershell.exe'}, (error, stdout, stderr) => {
        if(error || stderr){
            reject(error || stderr);
        }
        desktopPath = stdout.toString().trim();
        resolve();
    })
});
```

**Comment by sysrage** [comment]
> You need to change the exec to execSync for it to work. The regular exec works with a callback and doesn't return anything from the stdout. You can also do something like:

Nooo! This is already an async function so don't lock up the event loop! Instead of `process.execSync()` use `const exec = util.promisify(require('child_process').exec)` and `const { stdout, stderr } = await exec(...)`.

**Comment by TheJaredWilcurt** [comment]
> You need to change the exec to execSync for it to work. The regular exec works with a callback and doesn't return anything from the stdout.

The code already uses execSync and works as expected. I think you were just reading it wrong.

* * *

> Nooo! This is already an async function so don't lock up the event loop! Instead of process.execSync() use const exec = util.promisify(require('child_process').exec) and const { stdout, stderr } = await exec(...)

First off, see the README:

This library is completely synchronous

Secondly, the code you placed is not compatible with older versions of Node. This is a very general library, meant to solve a common issue in a cross-platform way. As such, compatibility is very important. It works on Node 4+, Windows XP+, Linux, and OSX.

* * *

The only thing left to do in this PR is update the unit tests to mock out the powershell call and to write tests for XP/Vista and 7+.

**Comment by StoneyEagle** [comment]
> > You need to change the exec to execSync for it to work. The regular exec works with a callback and doesn't return anything from the stdout.
> 
> The code already uses execSync and works as expected. I think you were just reading it wrong.
> 
> > Nooo! This is already an async function so don't lock up the event loop! Instead of process.execSync() use const exec = util.promisify(require('child_process').exec) and const { stdout, stderr } = await exec(...)
> 
> First off, see the README:
> 
> This library is completely synchronous
> 
> Secondly, the code you placed is not compatible with older versions of Node. This is a very general library, meant to solve a common issue in a cross-platform way. As such, compatibility is very important. It works on Node 4+, Windows XP+, Linux, and OSX.
> 
> The only thing left to do in this PR is update the unit tests to mock out the powershell call and to write tests for XP/Vista and 7+.

Sorry I don't think about older versions of windows or node.

**Comment by TheJaredWilcurt** [comment]
**Update:** I found a much simpler/less convoluted approach that goes by feature detection instead of assumption of feature based on OS version. Now I just need to mock out `which` finding and not finding powershell, and then powershell returning and failing to return the desktop for the unit tests.

@StoneyEagle The older version was in reference to Brian's code. Though in retrospect your async example also contains features that are not present in the older versions of Node.

@sysrage I'm not against this repo being `async` and `sync`. It's just out of scope for this PR. No one has complained about it not being async before. Testing on Windows 7, it pretty consistently takes 65ms to run, or 330ms if it needs to run powershell to find the desktop. With that performance decrease I think an `async` mode is justifed. If you're interested in adding this functionality, I'd be happy to review a PR, but you'll be limited to ES5 callbacks for asynchronous coding in order to continue support for older versions of Node.js and OS's. Also I'd like to maintain 100% test coverage in this repo as it's basically "feature complete" at this point.

**Comment by StoneyEagle** [comment]
If it's not going to work out it's not, I just suggested a way that works for me.
You can add this to the readme for other users that use OneDrive sync with recent Windows and NodeJS versions.
If it's too much hassle to make this work, just close the issue.

File: tests/testHelpers.js [hunk 1]
```diff
@@ -145,6 +168,10 @@ const testHelpers = {
       'C:/file.ext': 'text',
       'C:/folder': {},
       'C:\\Program Files\\DUMMY\\app.exe': windowsExecutable,
+      'C:\\Program Files\\DUMMY\\powershell.exe': windowsExecutable,
```
Comment (TheJaredWilcurt) [inline tests/testHelpers.js:171]:
```suggestion
```

**Pull request merged**